### PR TITLE
Smarter image processing and skeleton loader

### PR DESCRIPTION
**Pull request by fairnightzz**

### CODE REVIEW

File: src/utils/requests.tsx
Comment (fairnightzz) [inline src/utils/requests.tsx]:
```suggestion
```

**Comment by MrPicklePinosaur** [comment]
did you have to branch off #11 for this MR

makes reviewing it a bit weird since i also see changes from the other MR. might be better to merge this branch into `local-storage` when creating MR

also u should delete the branches after merging

**Comment by MrPicklePinosaur** [comment]
by MR i mean PR

File: src/pages/Reader.tsx
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx]: why not `string[]`

File: src/utils/requests.tsx
Comment (MrPicklePinosaur) [inline src/utils/requests.tsx:26]: can chapter.name be null for any reason?

**Review by MrPicklePinosaur — APPROVED** [review]
lgtm more or less

File: src/utils/requests.tsx
Comment (fairnightzz) [reply src/utils/requests.tsx:26]: no because that would mean you have no file name (i think)

**Pull request merged**
